app/src/main/java/com/faskn/app/weatherapp/core/BaseFragment.kt

```diff
@@ -5,58 +5,45 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.annotation.LayoutRes
-import androidx.databinding.DataBindingComponent
 import androidx.databinding.DataBindingUtil
 import androidx.databinding.ViewDataBinding
 import androidx.databinding.library.baseAdapters.BR
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
 import androidx.navigation.Navigation
-import dagger.android.AndroidInjection
 
 /**
  * Created by Furkan on 2019-10-16
  */
-
 abstract class BaseFragment<VM : BaseViewModel, DB : ViewDataBinding>(
-    @LayoutRes val layout: Int,
+    @LayoutRes layoutResId: Int,
     viewModelClass: Class<VM>
-) : Fragment() {
+) : Fragment(layoutResId) {
 
-    open lateinit var binding: DB
-    lateinit var dataBindingComponent: DataBindingComponent
-    private fun init(inflater: LayoutInflater, container: ViewGroup) {
-        binding = DataBindingUtil.inflate(inflater, layout, container, false)
-        binding.lifecycleOwner = viewLifecycleOwner
-        binding.setVariable(BR.viewModel, viewModel)
-    }
+    protected lateinit var binding: DB
+        private set
 
-    open fun init() {}
-
-    private val viewModel by lazy {
-        (activity as? BaseActivity<*, *>)?.viewModelProviderFactory?.let {
-            ViewModelProvider(
-                this,
-                it
-            ).get(viewModelClass)
+    private fun init(view: View) {
+        binding = DataBindingUtil.bind(view)!!
+        with(binding) {
+            setVariable(BR.viewModel, viewModel)
+            lifecycleOwner = viewLifecycleOwner
         }
     }
 
-    open fun onInject() {}
+    open fun init() {}
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        AndroidInjection.inject(activity)
-        super.onCreate(savedInstanceState)
+    protected val viewModel: VM by lazy {
+        ViewModelProvider(this).get(viewModelClass)
     }
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        init(inflater, container!!)
+        init(super.onCreateView(inflater, container, savedInstanceState)!!)
         init()
-        super.onCreateView(inflater, container, savedInstanceState)
         return binding.root
```

Comment:
I think we can simplify this logic to something like;

```kotlin
binding = DataBindingUtil.inflate(inflater, layout, container, false)
binding.run {
    lifecycleOwner = this@BaseFragment
    setVariable(BR.viewModel, viewModel)
}
return binding.root
```

We can get rid of `init` function.

Reply:
Done. 
97c845a